Source/Core/VideoCommon/HiresTextures.cpp

```diff
@@ -135,25 +144,67 @@ void HiresTexture::Update()
       }
     }
 
-    s_textureCacheAbortLoading.Clear();
+    s_textureCacheAbortPrefetching.Clear();
     s_prefetcher = std::thread(Prefetch);
   }
 }
 
 void HiresTexture::Prefetch()
 {
+  // prefetch() is running in a tight loop, and it's using that to allow "at least" 1 Search()
+  // run per cycle
   Common::SetCurrentThreadName("Prefetcher");
 
-  size_t size_sum = 0;
-  size_t sys_mem = Common::MemPhysical();
-  size_t recommended_min_mem = 2 * size_t(1024 * 1024 * 1024);
-  // keep 2GB memory for system stability if system RAM is 4GB+ - use half of memory in other cases
-  size_t max_mem =
-      (sys_mem / 2 < recommended_min_mem) ? (sys_mem / 2) : (sys_mem - recommended_min_mem);
-  u32 starttime = Common::Timer::GetTimeMs();
-  for (const auto& entry : s_textureMap)
+  double prefetch_start_time_total = 0;
+  double prefetch_stop_time_total = 0;
+  constexpr double CULLED_PERCENT_DEFAULT = 30.0;
```

Comment:
This will lower the amount of RAM that Dolphin can use for prefetching by default on systems with over 6.67 GiB RAM. Is that really a good idea?

Reply:
Most likely not final numbers, as this hasn't been discussed yet definitively, ballpark was to leave at least 2-3 GB on something like 24GB system.

Reply:
See, if you want to risk it tighter, with lower headroom, then warning messages become tempting again. Also where did you get specific 6.67 GiB from ?

Reply:
2 GiB (the amount of space not usable for prefetching in master if you have 4 GiB RAM or more) divided by 0,3 (your new limit) gives 6.67 GiB.

Reply:
Oh, but we agreed to remove that whole thing and just replace it with a simple "leave X percent alone from system max" with INI adjustability and a low+high bound hard limit. This was all talked about on IRC long ago before I even wrote a single line, this is all dejavu btw, I have already gone into the log archives twice and we had this discussion once.